Update dzPresence.lua to work with Windows Domoticz

lamama1234 opened the pull request.

lamama1234 committed:
Update dzPresence.lua to work with Windows Domoticz

Nateonas commented:
Thanks you very much! I see lots of changes, but they are only spaces, returns and tabs, right?

The only _real_ change is in line 84, replacing
`local shellcommand =               pingcommand .. ' ' .. ip          -- e.g. ping -q -w 5 192.168.1.1`

with:
`local shellcommand = 'cmd /C "' .. pingcommand .. " " .. ip .. '"'   -- Using cmd /C to allow shell features like | to work (if needed in the future)`

Did you make any other changes?

If not, then I _did_ anticipate that another user would want to use a different shellcommand, see line 46:
`local pingcommand  = 'ping -q -w 5' `

I think that (just) customizing the user defaults would also do the job:
`local pingcommand  = 'cmd /C ping -q -w 5' `

Would you agree?

lamama1234 commented:
Hi,

Basically it's the ping command + the block with:

`local pingSuccess = string.match(item.data, "bytes=%d+")`

I changed that logic because Windows ping counts any ICMP reply — even errors like "host unreachable" — as a 'received' packet.

There are also a few places i added on/off verification prior to switch state function calls, e.g.

`if domoticz.devices(any_idx).state ~= "Off"`

Thanks!
